### ISSUE TITLE
Very slow rendering of graphs with multiple graphs / plotly

### ISSUE THREAD

**Issue by luck02** [jupyterlab/jupyterlab#5738]
I'm wondering if anyone else is seeing this behaviour.

I am iteratively creating graphs with relatively modest sized datasets (At most 150 rows).  It's all summary data, 3-5 columns * up to 150 rows.

After about 3-4 graphs created in a given page the rendering takes a huge hit in terms of speed.  The page seems to freeze up and with 15+ graphs on a given notebook it can take up to 20 minutes to fully render, once it is fully rendered it's reasonably performant (other than being very slow to switch tabs).

using https://hub.docker.com/r/jupyter/all-spark-notebook/tags/ on latest I think that's (14fdfbf9cfc1) 

And plotly version: @jupyterlab/plotly-extension v0.18.1

```
root@47712b0e71c0:~# jupyter --version
4.4.0

root@47712b0e71c0:~# jupyter labextension list
JupyterLab v0.35.4
Known labextensions:
   app dir: /opt/conda/share/jupyter/lab
        @jupyter-widgets/jupyterlab-manager v0.38.1  enabled  OK
        @jupyterlab/hub-extension v0.12.0  enabled  OK
        @jupyterlab/plotly-extension v0.18.1  enabled  OK
        jupyterlab_bokeh v0.6.3  enabled  OK
```
I'm mostly wondering if my use case (multiple distinctly drawn graphs in one notebook is a common or extreme edge case.  And really, is anyone else seeing this or just me?

**Comment by aquirdTurtle** [comment]
Not sure about exact numbers but I've also seen similar behavior and just went back to using matplotlib for the majority of my plots. Would definitely be nice if the notebook could handle more plotly though!

**Comment by jasongrout** [comment]
Is this a plotly issue or a notebook issue? Is it a problem to have this many large plotly graphs on a webpage outside of jlab?

**Comment by luck02** [comment]
@jasongrout I'm actually not sure.  I'll open an issue in plotly as well.

I'm unsure if it's just me, or if it's a general issue or if others even have enough charts on a single notebook.  

Opened issue here: https://community.plot.ly/t/plotly-notebook-very-slow-render-with-many-graphs/16861  If I get relevant information I'll bring it back here.

**Comment by sbowman-mitre** [comment]
I'll just jump in a say I definitely see this too. Everything starts grinding once I have a few plotly graphs in any one notebook. Even click events (to edit a snippet of code) takes a long time to respond.

**Comment by jasongrout** [comment]
@jonmmease - I see you are working on something to try to address this in the plotly extension. Any idea why people are seeing this slowdown with many plotly graphs?

**Comment by jonmmease** [comment]
> Any idea why people are seeing this slowdown with many plotly graphs?

I don't have any specific theories at this point beyond the size of the DOM growing large.

> I see you are working on something to try to address this in the plotly extension

We haven't done anything in the plotly renderer extension at this point.  The one approach which has helped some folks is the switch to the `iframe` renderer as described in https://community.plot.ly/t/plotly-notebook-very-slow-render-with-many-graphs/16861/7.  This stores each figure in a separate HTML file and displays them in JupyterLab through an iframe.

**Comment by sbowman-mitre** [comment]
Let me jump in & say that I've seen a definite scrolling improvement by managing dependency versions more carefully. I updated to plotly 3.9, which seemed to help some. Then I noticed that jupyterlab 0.35.6 was just released and newer than 1.0.0a, so I rolled back to that. I also switched to @jonmmease suggesting of using iframes, which I do think that made a huge difference. 

At the same time, all is not well. I still experience large delays when clicking in a cell and waiting for a response. And anything that occurs in the cell is crazy slow, which effectively kills editing code. So, I still find I can't continue to develop notebooks that already contain multiple plotly graphs.
